server/src/app.module.ts

```diff
@@ -1,29 +1,58 @@
 import { MiddlewareConsumer, Module, NestModule, RequestMethod } from '@nestjs/common';
-
+import { APP_FILTER } from '@nestjs/core';
+import { AllExceptionsFilter } from './@common/filters/all-exceptions.filter';
+import { appConfig } from 'config/config';
 import { AuthModule } from './modules/auth/auth.module';
 import { JsonBodyMiddleware } from './middlewares/json-body.middleware';
 import { RawBodyMiddleware } from './middlewares/raw-body.middleware';
 import { HTTPLoggerMiddleware } from './middlewares/http-logger.middleware';
 import { MapsModule } from './modules/maps/maps.module';
 import { UsersModule } from './modules/users/users.module';
-import { HealthModule } from './modules/health/health.module';
-import { MetricsModule } from './modules/metrics/metrics.module';
 import { UserModule } from './modules/user/user.module';
+import { SentryModule } from './@common/sentry/sentry.module';
+
+import * as Sentry from '@sentry/node';
 
 @Module({
-    imports: [UsersModule, MapsModule, UserModule, AuthModule, HealthModule, MetricsModule]
+    imports: [
+        UsersModule,
+        MapsModule,
+        UserModule,
+        AuthModule,
+        SentryModule.forRoot({
+            dsn: appConfig.sentry.dsn,
+            tracesSampleRate: 1.0,
+            debug: appConfig.sentry.debug,
+            environment: process.env.NODE_ENV || 'development'
+        })
+    ],
+    providers: [
+        {
+            provide: APP_FILTER,
+            useClass: AllExceptionsFilter
```

Comment:
Same as above

Reply:
UseValue is used when you want to provide a static Object, use class is used when you want to provide a static Service or similar that has a constructor that needs to be run for the service to be useful.

https://medium.com/@paynoattn/use-usevalue-if-your-stub-is-a-object-useclass-if-your-stub-is-a-class-88ba6e1574f0
https://offering.solutions/blog/articles/2018/08/17/using-useclass-usefactory-usevalue-useexisting-with-treeshakeable-providers-in-angular/